### DISCUSSION TITLE
Ability auto reserve the name of the person throughout the world while a kid is born

### DISCUSSION THREAD

**Discussion by ghost** [mosip/mosip#4] [Ideas]
Ability auto trademark and procure domain(country or union TLD) the name of the person through out the world while a kid is born or let the parents/guardians know they are naming of baby infringes name of company/trademark domain/already exists a person with same name and advise not use it or perhaps help generate unique  names leveraging traditional naming patterns especially name of person and **not** necessarily tag along family name/father's name/religion name/local area name/ husband name as if their property another person instead promote freedom to live with their own name and make life in this society in future.

**Comment by gsasikumar** [comment]
@prnam I am unable to understand the question. Can you please give some examples of what you are looking for?

**Reply by ghost** [reply]
@gsasikumar,
1. A idea/feature request to have parents or individuals to get trademark, domain name .in, when a person is born or existing person be able easily get it than separately request it. Also, avoid naming the baby based on exiting trademark and domain.

2. A idea/feature request to recommend not use name which are commonly used when there is more than 1 name already in national ID database within a country or around the world.

3. A idea/feature request to recommend parents that child does not necessarily need last name.

**Reply by gsasikumar** [reply]
@prnam yes looks great. But as of now our focus is it provide ID and enable more people using it offline. But as a community if you are interested in building this then we could offer some ideas of this can be done and also we could host this in our market place. Please let me know your interest. Also most of our discussion are active in https://community.mosip.io . Github discussions are not actively monitored.

**Comment by ghost** [comment]
> @prnam yes looks great. But as of now our focus is it provide ID and enable more people using it offline. But as a community if you are interested in building this then we could offer some ideas of this can be done and also we could host this in our market place. Please let me know your interest. Also most of our discussion are active in https://community.mosip.io . Github discussions are not actively monitored.

@gsasikumar - I am working on couple of open-source projects and one of them is [Patashale™](https://www.github.com/patashale) which we have applied for non profit organisation. So at the moment I cannot pick it up, probably `Code4GovTech` NPO can.
